flake.nix

```diff
@@ -0,0 +1,184 @@
+{
+  description = "Compiling to categories";
+
+  nixConfig = {
+    ## https://github.com/NixOS/rfcs/blob/master/rfcs/0045-deprecate-url-syntax.md
+    extra-experimental-features = ["no-url-literals"];
+    extra-trusted-public-keys = [
+      "cache.garnix.io:CTFPyKSLcx5RMJKfLo5EEPUObbA78b0YQ2DTCJXqr9g="
+    ];
+    extra-trusted-substituters = ["https://cache.garnix.io"];
+    ## Prevent the Nix GC from removing current dependencies from the store.
+    keep-failed = true;
```

Comment:
I think we had better set these `keep-*` configurations under users' preference, not as flake defaults.